src/OSVRTrackedController.cpp

```diff
@@ -0,0 +1,515 @@
+    axis_state.y = static_cast<float>(analog_interface->y);
+
+    vr::VRServerDriverHost()->TrackedDeviceAxisUpdated(self->objectId_, analog_interface->axisIndex, axis_state);
+}
+
+void OSVRTrackedController::controllerYAxisCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_AnalogReport* report)
+{
+    if (!userdata)
+        return;
+
+    auto* analog_interface = static_cast<AnalogInterface*>(userdata);
+    OSVRTrackedController* self = analog_interface->parentController;
+
+    analog_interface->y = report->state;
+
+    vr::VRControllerAxis_t axis_state;
+    axis_state.x = static_cast<float>(analog_interface->x);
+    axis_state.y = static_cast<float>(analog_interface->y);
+
+    vr::VRServerDriverHost()->TrackedDeviceAxisUpdated(self->objectId_, analog_interface->axisIndex, axis_state);
+}
+
+const char* OSVRTrackedController::GetId()
+{
+    /// @todo When available, return the actual unique ID of the HMD
+    return name_.c_str();
+}
+
+void OSVRTrackedController::configure()
+{
+    configureProperties();
+}
+
+void OSVRTrackedController::configureProperties()
+{
+
+    propertyContainer_ = vr::VRProperties()->TrackedDeviceToPropertyContainer(this->objectId_);
+    // Additional properties from nolo's osvr tracked device
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_TrackingSystemName_String, "NoloVR");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_ManufacturerName_String, "LYRobotix");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_TrackingFirmwareVersion_String, "0.1.0");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_HardwareRevision_String, "0.1.0");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_AllWirelessDongleDescriptions_String, "");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_ConnectedWirelessDongle_String, "");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_Firmware_ManualUpdateURL_String, "");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_Firmware_ProgrammingTarget_String, "");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_DriverVersion_String, "0.1.0");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_AttachedDeviceId_String, "3000");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_ModeLabel_String, "");
+    //vr::VRProperties()->SetBoolProperty(propertyContainer_, vr::Prop_WillDriftInYaw_Bool, true);
+    //vr::VRProperties()->SetBoolProperty(propertyContainer_, vr::Prop_DeviceIsWireless_Bool, false);
+    vr::VRProperties()->SetInt32Property(propertyContainer_,  vr::Prop_DeviceClass_Int32,       static_cast<int32_t>(deviceClass_));
+    /*
+    vr::VRProperties()->SetInt32Property(propertyContainer_,  vr::Prop_Axis0Type_Int32,         static_cast<int32_t>(analogInterface_[0].axisType));
```

Comment:
I agree and its done in the registerTrigger, registerTouchpad, etc functions.  See
`void OSVRTrackedController::registerTrackpad(int id, const std::string& path){
    analogInterface_[id].analogInterfaceX = context_.getInterface(path + "/x");
    analogInterface_[id].analogInterfaceY = context_.getInterface(path + "/y");

    if (analogInterface_[id].analogInterfaceX.notEmpty()) {
        analogInterface_[id].axisIndex        = id;
        analogInterface_[id].axisType         = vr::EVRControllerAxisType::k_eControllerAxis_TrackPad;
	analogInterface_[id].parentController = this;
	analogInterface_[id].analogInterfaceX.registerCallback(&OSVRTrackedController::controllerXAxisCallback, &analogInterface_[id]);
        propertyContainer_                    = vr::VRProperties()->TrackedDeviceToPropertyContainer(this->objectId_);
        vr::VRProperties()->SetInt32Property(propertyContainer_, vr::Prop_Axis0Type_Int32, static_cast<int32_t>(analogInterface_[id].axisType));
    } else {
	analogInterface_[id].analogInterfaceX.free();
    }

    if (analogInterface_[id].analogInterfaceY.notEmpty()) {
        analogInterface_[id].axisIndex        = id;
        analogInterface_[id].axisType         = vr::EVRControllerAxisType::k_eControllerAxis_TrackPad;
	analogInterface_[id].parentController = this;
        analogInterface_[id].analogInterfaceY.registerCallback(&OSVRTrackedController::controllerYAxisCallback, &analogInterface_[id]);
        propertyContainer_                    = vr::VRProperties()->TrackedDeviceToPropertyContainer(this->objectId_);
        vr::VRProperties()->SetInt32Property(propertyContainer_, vr::Prop_Axis0Type_Int32, static_cast<int32_t>(analogInterface_[id].axisType));
    } else {
        analogInterface_[id].analogInterfaceY.free();
    }
}`